### PR TITLE
Bug 1828698 - Refactor `DropdownMenu` into a use case Composable

### PR DESCRIPTION
**Pull request by MozillaNoah**
This ticket started out as just a corner radius bump, but I opted to do the initial refactoring for making this dropdown one of our "use case" Composables, a la `Button` and `PrimaryButton`. I'll open tickets for the other two use cases outlined in the [menu figma](https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?type=design&node-id=1808-8898&t=774NQxZXkf7v9NZD-0) after this lands, assuming this is the strategy we agree on.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1828698

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt [hunk 1]
```diff
@@ -76,6 +85,33 @@ fun DropdownMenu(
...
+ * Dropdown menu for presenting context-specific actions.
+ *
+ * @param menuItems List of items to be displayed in the menu.
+ * @param showMenu Whether or not the menu is currently displayed to the user.
+ * @param onDismissRequest Invoked when user dismisses the menu or on orientation changes.
+ * @param modifier Modifier to be applied to the menu.
+ * @param offset Offset to be added to the position of the menu.
+ */
+@Composable
+fun ContextualDropdownMenu(
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:98]:
```suggestion
fun ContextDropdownMenu(
```

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:98]: Oh I see it's called Contextual in the figma. Maybe we should be trying to sway @cwzilla into renaming it to Context dropdown menu.

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:98]: Perhaps `ContextualMenu` for the time being?

File: fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt [hunk 2]
```diff
@@ -21,21 +27,24 @@ import androidx.compose.ui.platform.testTag
...
  * @param menuItems List of items to be displayed in the menu.
  * @param showMenu Whether or not the menu is currently displayed to the user.
+ * @param cornerShape Shape to apply to the corners of the dropdown.
  * @param onDismissRequest Invoked when user dismisses the menu or on orientation changes.
  * @param modifier Modifier to be applied to the menu.
  * @param offset Offset to be added to the position of the menu.
  */
 @Composable
-fun DropdownMenu(
+private fun DropdownMenu(
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:44]: I wonder if this naming will overlap with the actual Dropdown Menu component in figma?

**Review by gabrielluong — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:98]: `ContextualMenu` works for me, and I agree with keeping it succinct.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:44]: hrmm, good point. Maybe just `Menu`?

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:44]: `Menu` makes sense given it's the menu component in the figma. I would also suggest renaming the file name as well to match.

Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/DropdownMenu.kt:44]: good catch on the filename

**Comment by MozillaNoah** [comment]
Opened this [ticket for the main menu](https://bugzilla.mozilla.org/show_bug.cgi?id=1834654), and this [ticket for the sub menu](https://bugzilla.mozilla.org/show_bug.cgi?id=1834655).

**Pull request merged**